osutil/group.go

```diff
@@ -79,7 +79,22 @@ func getent(database, name string) (uint64, error) {
 var findUidNoGetentFallback = func(username string) (uint64, error) {
 	myuser, err := user.Lookup(username)
 	if err != nil {
-		return 0, err
+		// XXX: when using cgo, sometimes getpwnam_r will return ESRCH, which
+		// is not currently recognized as an UnknownUserError type of error by
+		// golang's implementation of user.LookupGroup, so to work around that,
+		// we treat all non-nil errors here as UnknownUserError, as actually
+		// the man page for getpwnam_r says:
+		//
+		// > It does not call "not found" an error, hence does not specify what
+		// > value errno might have in this situation. But that makes it
+		// > impossible to recognize errors.
+		//
+		// so we really have no way to tell what is an unknown group error and
+		// what is some other kind of error
+		// if there is a real problem finding the user then presumably other
+		// things will fail upon trying to create the user, etc. which will give
+		// more useful and specific errors
+		return 0, user.UnknownUserError(username)
```

Comment:
I can almost convince myself it makes sense. The lookup failed, so the user/group is effectively unknown.

I've looked at the code that may be affected and it's seems that only snapshots care about those particular kinds of errors. It does a look by uid, but AFAICT it will eventually hit nsswitch (due to getpwuid_r). If I'm reading this right, snapshotstate.Save() can actually create a snapshot where otherwise it would not create one due to errors from user.LookupId() bubbling up. I don't think it's a big problem though.

Reply:
I looked at the snapshot case and I agree I don't think it's a big problem, but note that the snapshot code doesn't use osutil.FindUid so it won't hit the code here. As such, I changed the code in snapshot and added this same comment there too